mlir/utils/performance/analysis/compileAndCollectTuningData.py

```
@@ -0,0 +1,604 @@
+    timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+
+    # Compile the config
+    debug_output = compile_config(config, perf_config, operation, binaries,
+                                  timestamp)
+
+    # Parse the results from the compiled config
+    results = parse_results(debug_output)
+
+    # Calculate occupancy using the method in testing_metrics.py
+    [M, N, G, MPerBlock, NPerBlock,
+        MNPerWave, minNumWaves, splitKFactor] = \
+                                gatherOccupancyParameters(config,
+                                                          perf_config,
+                                                          test_args,
+                                                          operation)
+    # If any of the parameters are None, we cannot calculate occupancy
+    if None in [M, N, G, MPerBlock, NPerBlock,
+                MNPerWave, minNumWaves, splitKFactor]:
+        print("Warning: Could not gather all parameters for occupancy "
+              "calculation for config. Skipping occupancy calculation.")
+        results.occupancy = None
+    else:
+        results.occupancy = calculateOccupancy(M, N, G, MPerBlock, NPerBlock,
+                                               MNPerWave, minNumWaves,
+                                               splitKFactor)
+
+    # Clean up temporary debug file
+    try:
+        if os.path.exists(debug_output):
+            os.remove(debug_output)
+    except Exception as e:
+        print(f"  Warning: Could not remove {debug_output}: {e}")
+
+    return results
+
+def write_results_to_tsv(results, configs):
+    """
+    Write the collected tuning data results to a tsv file.
+    
+    Args:
+        results: List of tuning data dictionaries
+        configs: List of original configuration dictionaries
+        output_file: Path to the output tsv file
+    """
+    if not results:
+        print("No results to write")
+        return
```

as Djordje suggested, let's propagate an error code or exception so main() can return 1